controllers/nodefeaturediscovery_status.go

```diff
@@ -0,0 +1,525 @@
+	degradedCondition := r.getDegradedConditions(reason, message)
+	if err := r.updateStatus(nfd, degradedCondition); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{Requeue: true}, nil
+}
+
+// updateProgressingCondition is used to mark a given resource as "progressing" so
+// that the reconciler can take steps to rectify the situation.
+func (r *NodeFeatureDiscoveryReconciler) updateProgressingCondition(nfd *nfdv1.NodeFeatureDiscovery, reason, message string) (ctrl.Result, error) {
+	progressingCondition := r.getProgressingConditions(reason, message)
+	if err := r.updateStatus(nfd, progressingCondition); err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{Requeue: true}, nil
+}
+
+// getAvailableConditions returns a list of conditionsv1.Condition objects and marks
+// every condition as FALSE except for conditionsv1.ConditionAvailable so that the
+// reconciler can determine that the resource is available.
+func (r *NodeFeatureDiscoveryReconciler) getAvailableConditions() []metav1.Condition {
+	now := time.Now()
+	return []metav1.Condition{
+		{
+			Type:               ConditionAvailable,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionUpgradeable,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionProgressing,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionDegraded,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+	}
+}
+
+// getDegradedConditions returns a list of conditions.Condition objects and marks
+// every condition as FALSE except for conditions.ConditionDegraded so that the
+// reconciler can determine that the resource is degraded.
+func (r *NodeFeatureDiscoveryReconciler) getDegradedConditions(reason string, message string) []metav1.Condition {
+	now := time.Now()
+	return []metav1.Condition{
+		{
+			Type:               ConditionAvailable,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionUpgradeable,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionProgressing,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionDegraded,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.Time{Time: now},
+			Reason:             reason,
+			Message:            message,
+		},
+	}
+}
+
+// getProgressingConditions returns a list of conditions.Condition objects and marks
+// every condition as FALSE except for conditions.ConditionProgressing so that the
```

Comment:
```suggestion
// getProgressingConditions returns a list of Condition objects and marks
// every condition as FALSE except for ConditionProgressing so that the
```